### DISCUSSION TITLE
filter/search for missing data

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/clomonitor#391] [Q&A] [answered]
Hi -- is there a way to filter projects by missing info/data?

For example, I'd like to see a list of all the CNCF projects that don't yet have a website, is there a way to filter or search for that?

**Comment by tegioz** [comment]
Hi @nate-double-u 👋 

Not yet, but we've been discussing the best way to add support for this recently as we think it could be really helpful 🙂

This is somehow related to the links in graphs you mentioned recently in #379.

CLOMonitor's main page is designed around projects: we display a list of them, it's possible to filter them, click on them to view more details, etc. However, checks are done at the *repository level*, not per project. A lot of projects have only one repository, but many others have multiple.

We'd love to add some extra filters that would allow you to get the **entities** that pass some checks and/or fail others, export the results to `csv`, etc. The problem is that if the entity was the **project**, what would it mean to pass or fail a check, that *all* of the repositories in the project would pass it? *any* of them? *none*? We are concerned this may not be super intuitive, but it'd be good to know what do you think. An alternative approach would be to add a section where the main entity was a **repository**. The problem just mentioned would disappear immediately given that checks are done per repository. But in this case, we don't know if we'd be complicating the user experience in CLOMonitor

Please let us know your thoughts as we'd love to move forward with this and any feedback is really appreciated 🙂

**Reply by nate-double-u** [reply]
> This is somehow related to the links in graphs you mentioned recently in #379.

Nope, just working through the site and asking about things as they come to me 🙂

This question is probably more closely related to my feature request https://github.com/cncf/clomonitor/issues/390.

**Reply by tegioz** [reply]
I probably did not explain myself well 🙂

While working on the clickable graphs issue, we went through all of the graphs in the stats page to check which would make more sense and be more useful. And we thought that creating links from the one below would be nice (i.e. take me to all projects passing / not passing the corresponding check clicked):

<img width="1439" alt="Screen Shot 2022-06-18 at 22 36 02" src="https://user-images.githubusercontent.com/1213902/174456348-73bdbc0a-e566-40ca-9590-80682577ad2e.png">

Adding links to the entries on the graph above would require the ability to filter projects that pass or fail some checks, which is what this discussion suggests. That's why I mentioned they were somehow related 😇 But to make this work we need to first work out the issue I introduced above as those checks are not really happening at the project level, but per repository. And many projects have multiple repositories registered.

Hope this makes sense now 🙂

**Reply by nate-double-u** [reply]
I've responded to this in the issue: https://github.com/cncf/clomonitor/issues/379#issuecomment-1159814769 (i think that filtering and graph links are linked, but different enough to be kept separate)

**Reply by nate-double-u** [reply]
The primary filter I'm looking for (that inspired this question) is the ability to see which projects don't have websites -- the Stats page is one possible way to find that info, but I think the search page (with its filters on the left) may be another way to find it. (This is part of why I think this thread is more closely related to https://github.com/cncf/clomonitor/issues/390, I believe that website data should be project level info, not repo level)

**Comment by jeefy** [comment]
Would it be possible to generate a CSV of all failed checks for everything? I'm thinking a schema like:
```csv
foundation, project, repo, check-name
cncf, smi, https://github.com/servicemeshinterface/smi-spec, adopters
```

I think the engineering time for that would be pretty quick, and then we can filter the data on our end to get the list of what to act on.

**Reply by tegioz** [reply]
Yes, that wouldn't be a problem. We were hoping to have something nicer in the UI and have the CSV export as an option. But having only this is definitely easier, it could be a temporary solution until we find a nice way to express this in the UI. We could even export not just the ones that failed, but all of them and add a `passed` field.

**Comment by tegioz** [comment, marked as answer]
This was addressed in #439 and #453 🙂

**Reply by nate-double-u** [reply]
This is awesome, thanks so much @tegioz & team!
